### ISSUE TITLE
Is the `itkMemoryProbesCollecterBaseTest` useful?  It fails on macOS, at least in some conditions

### ISSUE THREAD

**Issue by seanm** [InsightSoftwareConsortium/ITK#4818]
The `itkMemoryProbesCollecterBaseTest` fails on macOS, at least in some conditions, ex:

https://open.cdash.org/tests/1690707341

where it gives:

```
Freeing memory should result in less memory but it is 262216kB instead of 262164kB
```

I have a suspicion that this test is quite old, and being tripped up by modern OS memory management (ex: memory compression and other magic).

I wonder if this test really gives any value?

**Comment by seanm** [comment]
In fact, unless I've missed something in my searching, `itkMemoryProbe*` is only used in Examples and Tests...

**Comment by dzenanz** [comment]
Of course, that class is meant for testing and debugging. Perhaps increase the amount of memory being released? That should prevent it from being lost in the noise.

**Comment by seanm** [comment]
@dzenanz your idea worked! https://github.com/InsightSoftwareConsortium/ITK/pull/5152
